src/main/java/org/opensearch/security/privileges/SecurityIndexAccessEvaluator.java

```diff
@@ -71,122 +82,240 @@ public SecurityIndexAccessEvaluator(final Settings settings, AuditLog auditLog,
         this.systemIndexMatcher = WildcardMatcher.from(
             settings.getAsList(ConfigConstants.SECURITY_SYSTEM_INDICES_KEY, ConfigConstants.SECURITY_SYSTEM_INDICES_DEFAULT)
         );
-        this.systemIndexEnabled = settings.getAsBoolean(
+        this.protectedSystemIndexMatcher = WildcardMatcher.from(ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
```

Comment:
This is incorrect, if the security index is renamed by the configuration it will not be used, make the matcher from `this.securityIndex`, or just create the matcher inline as securityIndex is used throughout this file